core/trino-main/src/main/java/io/trino/server/ServerMainModule.java

```diff
@@ -442,9 +442,8 @@ protected void setup(Binder binder)
         binder.bind(NodeVersion.class).toInstance(new NodeVersion(nodeVersion));
         discoveryBinder(binder).bindHttpAnnouncement("trino")
                 .addProperty("node_version", nodeVersion)
+                .bindProperty("node_instance_id").toProvider(NodeInstanceId.class)
```

Comment:
This uses https://github.com/airlift/airlift/pull/1157

Reply:
I was struggling with that for a moment. Thanks for the clarification. I will have to rebuild airlift before trying out this PR.

Reply:
airlift released. code here is updated, so you can try it out @sajjoseph